README.md

```diff
@@ -1,16 +1,204 @@
+
+External links (not https://appwrite.io) should be opened in a new tab (`target="_blank"`) and have an HTML attribute of `rel="noopener"`.
+
+Links that are not inline:
+
+```html
+<p>
+<a href="/docs/environment-variables"><i class="icon-angle-circled-right"></i>Learn more about environment variables</a>
+</p>
+```
+
+### Referencing UI Elements
+When referencing buttons, tabs, and other interactive UI elements, use the full text found on the UI element in bold. For example **Create Project** or **Add Platform**.
+
+When referencing example copy text that appears or could appear on a piece of UI, use double quotes. For example, for each function listed in the **Functions** tab, you will see the runtime used by that function, such as "Dart 2.16" or "Node.js 16.0".
+
+### Referencing File Paths, URLs, and IDs
+When referencing file paths, URLs, IDs, or any text appropriate for monospace representation, use a span element with `class=tag` inline. For example, `<span class="tag">appwrite.json</span>` or `<span class="tag">https://[HOSTNAME_OR_IP]/v1</span>`.
+
+When using generic strings, use double quotes `"`. Double quotes express a string in **all** languages, but single quotes `'` in some languages like Java and C++ represent a **char**. This makes them difficult to copy and paste.
+
+### Procedural Instructions
+When describing steps to do something, especially UI workflows, use an ordered list. For example:
+
+```html
+<ol>
+    <li>Create a script to backups and restore your MariaDB Appwrite schema. Note that trying to backup MariaDB using a docker volume backup can result in a corrupted copy of your data. It is recommended to use MariaDB or MySQL built-in tools for this.</li>
+    <li>Create a script to backups and restore your InfluxDB stats. If you don’t care much about your server stats, you can skip this.</li>
+    <li>Create a script to backup Appwrite storage volume. There are many online resources explaining different ways to backup a docker volume. When running on multiple servers, it is very recommended to use an attachable storage point. Some cloud providers offer integrated backups to such attachable mount like GCP, AWS, DigitalOcean, and the list continues.</li>
+</ol>
+```
 
 ### Code Examples
 
+Use the following HTML structure to present code examples:
+
+```html
+<div class="ide margin-top-small" data-lang="ruby" data-lang-label="Ruby SDK">
+    <pre class="line-numbers"><code class="prism language-ruby" data-prism>#ruby code here...</code></pre>
+</div>
+```
+
+Code examples should require minimum modifications to be executed where possible (working and self-contained).
+
+As of writing this, these are the supported languages for code examples:
```

Comment:
Can we also include their IDs, that you should use in class and `data-lang`? Maybe even labels, so we stay consistent? It could be markdown table.